src/app/feed/feed-post-icon-row/feed-post-icon-row.component.html

```diff
@@ -1,152 +1,350 @@
       }"
+      class="position-relative cursor-pointer d-flex align-items-center pt-5px"
+      (mouseover)="debouncedToggleSelectReactionFunction(true)"
+      (mouseleave)="debouncedToggleSelectReactionFunction(false)"
     >
-      <div
-        [ngClass]="{ 'bg-danger': diamondDragCancel, show: diamondDragging }"
-        class="diamond-mobile-drag-instructions"
+      <div *ngIf="choosingReaction">
+        <ng-container *ngTemplateOutlet="reactionsPopTemplate"></ng-container>
+      </div>
+
+      <i-feather
+        [ngClass]="{
+          is_animating_ice: processedReaction,
+          'thumbs-up': postContent.PostEntryReaderState ? !postContent.PostEntryReaderState.LikedByReader : true,
+          'thumbs-up-fill': myReactions.length > 0
+        }"
+        name="thumbs-up"
+        class="feed-post-icon-row__icon thumbs-up"
+      ></i-feather>
+      <span *ngIf="!hideNumbers">{{ postReactionCounts.Total }}</span>
+    </div>
+
+    <ng-template #reactionSwiper>
+      <swipe-to-choose
+        [itemCount]="allowedReactions.length"
+        (selectItem)="sendReaction(allowedReactions[$event])"
+        (sendOne)="sendReaction(allowedReactions[0])"
       >
-        <p class="d-block">
-          {{ diamondDraggedText() }}
-        </p>
+        <ng-template #actionBtn>
+          <div
+            (click)="sendReaction(allowedReactions[0], $event)"
+            [ngClass]="{
+              'fc-blue': myReactions.length > 0
+            }"
+            class="position-relative cursor-pointer d-flex align-items-center"
+            (mouseover)="debouncedToggleSelectReactionFunction(true)"
+            (mouseleave)="debouncedToggleSelectReactionFunction(false)"
+          >
+            <div *ngIf="choosingReaction">
+              <ng-container *ngTemplateOutlet="reactionsPopTemplate"></ng-container>
+            </div>
+
+            <i-feather
+              [ngClass]="{
+                is_animating_ice: processedReaction,
+                'thumbs-up': postContent.PostEntryReaderState ? !postContent.PostEntryReaderState.LikedByReader : true,
+                'thumbs-up-fill': myReactions.length > 0
+              }"
+              name="thumbs-up"
+              class="feed-post-icon-row__icon thumbs-up"
+            ></i-feather>
+            <span *ngIf="!hideNumbers">{{ postReactionCounts.Total }}</span>
+          </div>
+        </ng-template>
+
+        <ng-template #item let-index>
+          <div
+            class="reactions-list__item reactions-list__item-mobile show"
+            [ngClass]="{
+              selected: hasUserReacted(allowedReactions[index]),
+              bouncing: processedReaction === allowedReactions[index],
+              inactive: processedReaction && processedReaction !== allowedReactions[index]
+            }"
+            (click)="sendReaction(allowedReactions[index], $event)"
+          >
+            <reactions-image [reaction]="allowedReactions[index]"></reactions-image>
+          </div>
```

Comment:
On mobile, the animation is not shown to simplify rendering since the popup gets immediately closed when you untap your finger